bddisasm/bdx86_decoder.c

```diff
@@ -3813,22 +3813,27 @@ NdGetEffectiveAddrAndOpMode(
     static const ND_UINT8 szLut[3] = { ND_SIZE_16BIT, ND_SIZE_32BIT, ND_SIZE_64BIT };
     ND_BOOL w64, f64, d64, has66;
 
-    if ((ND_CODE_64 != Instrux->DefCode) && !!(Instrux->Attributes & ND_FLAG_IWO64))
+    // Branchless form of (ND_CODE_64 != Instrux->DefCode) && !!(Instrux->Attributes & ND_FLAG_IWO64)
+    if (((ND_CODE_64 ^ Instrux->DefCode) * (Instrux->Attributes & ND_FLAG_IWO64)) != 0)
```

Comment:
I would expect the branch predictors in modern CPUs to be quite competent at predicting the branches here, since they operate on `INSTRUX` state that will very often be the same, allowing for a very high prediction rate.
I understand that these changes **may** remove some conditional moves or branches, but is it really worth it? At some point, one has to decide whether the (small) increase in performance is worth the less readable code. I don't think this is the case here. In addition, depending on the architecture, the multiplication may be a new source of overhead.

Reply:
Multiplication pipelines easier than branches. I wasn't entirely happy with this commit either but it did make it faster.

Reply:
The compiler may even ditch the multiplication altogether, and rely on conditional instructions (such as `CMOVcc` or `SETcc`), but this is not the point. The point is that it shouldn't really matter, since modern branch prediction should easily handle these branches with minimal overhead. These statements are not a performance bottleneck.